src/threadpool.c

```diff
@@ -261,6 +261,26 @@ static void init_once(void) {
   init_threads();
 }
 
+int uv_set_threadpool_size(unsigned int number) {
+  int r;
+  char num_str[12]; /* enough room for unsigned int and null terminator */
+
+  /* check if the threadpool is already initialized */
+  if (nthreads != 0)
```

Comment:
I wonder: would it be possible to enlarge the amount of threads while the pool is running?

Reducing it is probably hard / impossible, but adding new threads might be within the realm of possibility.

Reply:
I think that is out of the scope of the issue itself. I could work on that as well. It could be a mix between @santigimeno suggestion of a getter for this.

Reply:
I have a branch from 2015 that does just that: https://github.com/bnoordhuis/libuv/commits/auto-threadpool

Shrinking is done lazily. Threads terminate when they're idle and the actual thread count > desired thread count.

Reply:
@bnoordhuis  Do you think I should pivot this approach into something like yours instead?

Reply:
I'd say that's a good idea. https://github.com/libuv/libuv/pull/382 is the PR from back then, you will want to read through the discussion.

cc @saghul as a fyi. You had some comments back then; maybe you still do?